File: app/src/main/java/cn/ifafu/ifafu/data/http/APIManager.java
package cn.ifafu.ifafu.data.http;

import cn.ifafu.ifafu.data.http.service.WeatherService;
import cn.ifafu.ifafu.data.http.service.ZhengFangService;

public class APIManager {

    private static WeatherService weatherAPI;
    private static ZhengFangService zhengFangAPI;

    public static WeatherService getWeatherAPI() {
        if (weatherAPI == null) {
            weatherAPI = RetrofitManager.obtainService(WeatherService.class);
        }
        return weatherAPI;
    }

    public static ZhengFangService getZhengFangAPI() {
        if (zhengFangAPI == null) {
            zhengFangAPI = RetrofitManager.obtainService(ZhengFangService.class);
        }
        return zhengFangAPI;
    }

}

File: app/src/main/java/cn/ifafu/ifafu/data/http/APIManager.kt
package cn.ifafu.ifafu.data.http

import cn.ifafu.ifafu.data.http.elec.RetrofitFactory
import cn.ifafu.ifafu.data.http.service.WeatherService
import cn.ifafu.ifafu.data.http.service.XfbService
import cn.ifafu.ifafu.data.http.service.ZhengFangService

object APIManager {
    val weatherAPI: WeatherService by lazy {
        RetrofitManager.obtainService(WeatherService::class.java)
    }

    val zhengFangAPI: ZhengFangService by lazy {
        RetrofitManager.obtainService(ZhengFangService::class.java)
    }

    val xfbAPI: XfbService by lazy {
        RetrofitFactory.obtainService(XfbService::class.java)
    }

}